### PR TITLE
add batch client timeouts, set timeout in ci

### PR DESCRIPTION
**Pull request by danking**
This should prevent CI from getting killed by health checks if batch fails to respond in a timely manner.

### CODE REVIEW

File: batch/batch/api.py [hunk 1]
```diff
@@ -3,62 +3,109 @@
 import random
 import requests
 
+class API():
+    def __init__(self, timeout=None):
+        self.timeout = timeout
```
Comment (tpoterba) [inline batch/batch/api.py]: can you add a docstring here including the units for timeout?

File: batch/batch/api.py [hunk 2]
```diff
@@ -3,62 +3,109 @@
...
+    def cancel_job(url, job_id):
+        r = requests.post(url + '/jobs/{}/cancel'.format(job_id), timeout = self.timeout)
+        r.raise_for_status()
+        return r.json()
+
+    def create_batch(url, attributes):
+        d = {}
+        if attributes:
+            d['attributes'] = attributes
+        r = requests.post(url + '/batches/create', json = d, timeout = self.timeout)
```
Comment (tpoterba) [inline batch/batch/api.py]: PEP style prefers no spaces around the `=` in kwargs, but I don't really care.

So far I've tried to use this style for everything that isn't Hail annotate_... calls.

**Comment by cseed** [comment]
I'm pretty sure this is broken and the batch tests are going into an infinite loop.

**Comment by danking** [comment]
I changed the default timeout to 60s.

**Comment by danking** [comment]
I also switched the `api` argument to `BatchClient` to `default_api` rather than `None`. I wonder if that was somehow creating a muffled error message. It should have erred as soon as someone tried to make a request with `api=None`. Not sure why it looped forever instead.

**Pull request merged**